sphinx/application.py

```diff
@@ -326,12 +327,17 @@ def warn(self, message, location=None, prefix='WARNING: '):
 
         *prefix* usually should not be changed.
 
+        *type* and *subtype* are used to suppress warnings with :confval:`supress_warnings`.
+
         .. note::
 
            For warnings emitted during parsing, you should use
            :meth:`.BuildEnvironment.warn` since that will collect all
            warnings during parsing for later output.
         """
+        if is_suppressed_warning(type, subtype, self.config.suppress_warnings):
```

Comment:
I think it'd make more sense to refactor this into a `matches_warnspec` function (name to be bikeshed) and just `and self.config.suppress_warnings` it here.  A function which always depends on its third parameter doesn't sound too useful.